### PR TITLE
Avoid redundant and hard-coded support power cursors

### PR DESCRIPTION
**Pull request by Mailaender**
including some additional lint checking.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs [hunk 1]
```diff
@@ -63,8 +57,8 @@ public AirstrikePower(Actor self, AirstrikePowerInfo info)
 
 		public override void SelectTarget(Actor self, string order, SupportPowerManager manager)
 		{
-			if (info.UseDirectionalTarget)
-				self.World.OrderGenerator = new SelectDirectionalTarget(self.World, order, manager, Info.Cursor, info.DirectionArrowAnimation, info.DirectionArrowPalette);
+			if (info.UseDirectionalTarget && Info is DirectionalSupportPowerInfo directionalSupportPowerInfo)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs]: Is this needed? `AirstrikePowerInfo` explicitly inherits `DirectionalSupportPowerInfo` so it should be fine to pass it directly.

Comment (pchote) [reply OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs]: Or perhaps better to move this method into `DirectionalSupportPower`?

File: OpenRA.Mods.Common/Traits/SupportPowers/ParatroopersPower.cs [hunk 2]
```diff
@@ -81,8 +75,8 @@ public ParatroopersPower(Actor self, ParatroopersPowerInfo info)
 
 		public override void SelectTarget(Actor self, string order, SupportPowerManager manager)
 		{
-			if (info.UseDirectionalTarget)
-				self.World.OrderGenerator = new SelectDirectionalTarget(self.World, order, manager, Info.Cursor, info.DirectionArrowAnimation, info.DirectionArrowPalette);
+			if (info.UseDirectionalTarget && Info is DirectionalSupportPowerInfo directionalSupportPowerInfo)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/SupportPowers/ParatroopersPower.cs]: Likewise here.

**Review by RoosterDragon — APPROVED** [review]

File: OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddSpawnActorPowerBlockedCursor : UpdateRule
```
Comment (PunkPun) [inline OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs]: doesn't this need to be added to update path?

Comment (Mailaender) [reply OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs]: Added.

**Review by pchote — COMMENTED** [review]
Code changes LGTM but not tested.

File: OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		{
+			var spawnActorPower = actorNode.LastChildMatching("SpawnActorPower", false);
+			if (spawnActorPower == null)
+				yield break;
+
+			if (spawnActorPower.LastChildMatching("BlockedCursor", false) != null)
+				yield break;
+
+			var blockedCursorNode = new MiniYamlNodeBuilder("BlockedCursor", new MiniYamlBuilder("move-blocked"));
+			spawnActorPower.AddNode(blockedCursorNode);
```
Comment (PunkPun) [inline OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs]:
```suggestion
			foreach (var spawnActorPower in actorNode.ChildrenMatching("SpawnActorPower"))
			{
				var cursor = spawnActorPower.LastChildMatching("BlockedCursor");
				if (cursor != null && !cursor.IsRemoval())
					yield break;

				var blockedCursorNode = new MiniYamlNodeBuilder("BlockedCursor", new MiniYamlBuilder("move-blocked"));
				spawnActorPower.AddNode(blockedCursorNode);
			}
```

this should be a foreach. Also doesn't GrantExternalConditionPower also need the update rule?

**Review by PunkPun — COMMENTED** [review]
Otherwise LGTM

File: OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs (hunk 4 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/UpdateRules/Rules/20231010/AddSpawnActorPowerBlockedCursor.cs:33]: Why does `GrantExternalConditionPower` need an update rule?

**Review by PunkPun — APPROVED** [review]
LGTM

**Comment by PunkPun** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/402f14d7f1ce1b1a0baad3f78a2e9ae7ee3f87fc...b81057949dfead7ae125c46772107a500836ddb7)

**Pull request merged**